runtime/src/accounts_db.rs

```diff
@@ -2862,6 +3005,8 @@ impl AccountsDB {
             self.accounts_cache.set_max_flush_root(*root);
         }
 
+        self.squash_uncleaned_pubkeys(self.accounts_cache.fetch_max_flush_root());
```

Comment:
for the record, I even considered for the case of no `--accounts-db-caching-enabled`. it seems to work because squashing is optional. :)

Reply:
Yea, squashing is an optional memory-saving optimization.